cupy/_core/internal.pyx

```diff
@@ -53,10 +54,18 @@ cpdef inline tuple get_size(object size):
         )
         return ()
     if cpython.PySequence_Check(size):
-        return tuple(size)
-    if isinstance(size, int):
-        return size,
-    raise ValueError('size should be None, collections.abc.Sequence, or int')
+        try:
+            return tuple(size)
+        except TypeError:
+            # A numpy.ndarray unconditionally succeeds in PySequence_Check as
+            # it implements __getitem__, but zero-dim one is an unsized object
+            # and fails to make a tuple from it.
+            pass
+    try:
+        return operator.index(size),
```

Comment:
I suppose CuPy must reject `bool` and `np.bool_`.  Are they rejected somewhere else?
https://github.com/numpy/numpy/blob/2e2c4c20c745781719006e6110b6f1236c4e51f9/numpy/core/src/multiarray/conversion_utils.c#L918-L926

Reply:
Makes sense. Seems they are not rejected anywhere.